src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -790,7 +807,7 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
 
     proxy_cert = proxy_cert.replace('\\', r'\\\\')
 
-    if https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy and not auto_upgrade:
+    if https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy and not auto_upgrade and not container_registry_repository and not default_registry:
```

Comment:
What if the customer only passes --anonymous-pull? Is that scenario not allowed? What is the CLI experience for a pcr customer if he wants to enable anonymous pull on his pcr?

Reply:
I think we should also do some pcr parameter validations and throw appropriate errors. For eg: if user passes both  'container_registry_repository' and 'default_registry' together, or he passes 'container_registry_password' and 'anonymous_pull' together

Reply:
ack